sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionOwnership.cs

```diff
@@ -0,0 +1,104 @@
+
+        /// <summary>
+        ///   The name of the consumer group this partition ownership is associated with.
+        /// </summary>
+        ///
+        public string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   The identifier of the associated <see cref="EventProcessor" /> instance.
+        /// </summary>
+        ///
+        public string InstanceId { get; }
+
+        /// <summary>
+        ///   The identifier of the Event Hub partition this partition ownership is associated with.
+        /// </summary>
+        ///
+        public string PartitionId { get; }
+
+        /// <summary>
+        ///   TODO: purpose not clear yet.
+        /// </summary>
+        ///
+        internal long OwnerLevel { get; }
+
+        /// <summary>
+        ///   The offset of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />
+        ///   upon ownership update.
+        /// </summary>
+        ///
+        public long? Offset { get; }
+
+        /// <summary>
+        ///   The sequence number of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />
+        ///   upon ownership update.
+        /// </summary>
+        ///
+        public long? SequenceNumber { get; }
+
+        /// <summary>
+        ///   The date and time, in UTC, that the last update was made to this ownership.
+        /// </summary>
+        ///
+        public DateTimeOffset? LastModifiedTime { get; }
+
+        /// <summary>
+        ///   The entity tag needed to update this ownership.
+        /// </summary>
+        ///
+        public string ETag { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionOwnership"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The path of the specific Event Hub this partition ownership is associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group this partition ownership is associated with.</param>
+        /// <param name="instanceId">The identifier of the associated <see cref="EventProcessor" /> instance.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition this partition ownership is associated with.</param>
+        /// <param name="offset">The offset of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
+        /// <param name="sequenceNumber">The sequence number of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
+        /// <param name="lastModifiedTime">The date and time, in UTC, that the last update was made to this ownership.</param>
+        /// <param name="eTag">The entity tag needed to update this ownership.</param>
+        ///
+        public PartitionOwnership(string eventHubName,
```

Comment:
Do we want developers to be able to create this, or is this a class that we create internally?

Reply:
Good point. They should be created only before calling `IPartitionManager.ClaimOwnership`. Even though it's a user-implemented method, the instances should be created by us. I'll update it.